### ISSUE TITLE
vue/no-undef-components is expected to report error if component is type only

### ISSUE THREAD

**Issue by Amorites** [vuejs/eslint-plugin-vue#1963]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**  8.23.0
- **eslint-plugin-vue version:** 9.4.0
- **Node version:** 16.17.0
- **Operating System:** MacOS 12.5

**Please show your full configuration:**

```json5
module.exports = {
  extends: [
    // add more generic rulesets here, such as:
    // 'eslint:recommended',
    'plugin:vue/vue3-recommended',
    // 'plugin:vue/recommended' // Use this if you are using Vue.js 2.x.
  ],
  rules: {
    // override/add rules settings here, such as:
    // 'vue/no-unused-vars': 'error'
    'vue/no-undef-components': 'error',
  },
  parser: 'vue-eslint-parser',
  parserOptions: {
    extraFileExtensions: ['.vue'],
    parser: '@typescript-eslint/parser',
    ecmaVersion: 2022, // Allows for the parsing of modern ECMAScript features
    sourceType: 'module', // Allows for the use of imports
  },
};

```

**What did you do?**	

```vue
<script setup lang="ts">
import type HelloWorld from './components/HelloWorld.vue';
</script>

<template>
  <div>
    <a href="https://vitejs.dev" target="_blank">
      <img src="/vite.svg" class="logo" alt="Vite logo" />
    </a>
    <a href="https://vuejs.org/" target="_blank">
      <img src="./assets/vue.svg" class="logo vue" alt="Vue logo" />
    </a>
  </div>
  <HelloWorld msg="Vite + Vue" />
  <Foo></Foo>
</template>
```

**What did you expect to happen?**

` <HelloWorld msg="Vite + Vue" />` is expected to report no-undef-component error like `<Foo></Foo>`

**What actually happened?**

`<HelloWorld msg="Vite + Vue" />` doesn't trigger a eslint error

**Repository to reproduce this issue**

https://github.com/Amorites/eslint-plugin-vue-repro/blob/main/src/App.vue

**Comment by Amorites** [comment]
After doing some inspect, I found that it might be possible to use `importKind` to distinguish whether it's a type-only import or not.

With any luck, I can make a PR today.
